presto-main/src/main/java/com/facebook/presto/operator/SetBuilderOperator.java

```diff
@@ -203,12 +225,39 @@ public void addInput(Page page)
 
         Block sourceBlock = page.getBlock(setChannel);
         Page sourcePage = hashChannel.isPresent() ? new Page(sourceBlock, page.getBlock(hashChannel.get())) : new Page(sourceBlock);
-        channelSetBuilder.addPage(sourcePage);
+        unfinishedWork = channelSetBuilder.addPage(sourcePage);
+        processUnfinishedWork();
+        channelSetBuilder.updateMemoryReservation();
     }
 
+    /**
+     * This function always returns null because SetBuilderOperator is always the last operator.
```

Comment:
This function always returns null because `SetBuilderOperator` doesn't produce output and not just because  this operator is usually placed last on some list.

Remove this comment.

Reply:
done